lib/utils/job-utils/catalog-searcher.js

```diff
@@ -79,100 +79,212 @@ function searchCatalogDataFactory(
+            *                    "EID:Slt": "252:0",
+            *                    "Intf": "SAS",
+            *                    "Med": "HDD",
+            *                    "Model": "ST1200MM0088    ",
+            *                    "PI": "N",
+            *                    "SED": "N",
+            *                    "SeSz": "512B",
+            *                    "Size": "1.091 TB",
+            *                    "Sp": "U",
+            *                    "State": "Onln"
+            *                }
+            */
+            var driveDataList,
+                driveBaseInfoList = {},
+                matchedBaseInfo,
+                matchedBaseInfoKey;
+
+            _.forEach(physicalDiskCatalog.data.Controllers, function(controller){
+                if (controller['Command Status'].Controller.toString() === controllerId) {
+                    driveDataList = controller['Response Data'];
+                    return false;
+                }
             });
-            driveId.deviceIds = _.map(driveId.physicalDisks, function (disk) {
-                return disk.deviceId;
+
+            _.forEach(driveDataList, function(driveInfo, key){
+                if(_.isArray(driveInfo)) {
+                    driveBaseInfoList[key] = driveInfo[0];
+                }
             });
-            driveId.slotIds = _.map(driveId.physicalDisks, function (disk) {
-                // slotId : /c0/e252/s10
-                return '/c%d/e%d/s%d'.format(cid, disk.enclosureId, disk.slotId);
+
+            _.forEach(driveBaseInfoList, function(info, key){
+                if(info.DID.toString() === deviceId){
+                    matchedBaseInfoKey = key;
+                    return false;
+                }
             });
-            // OEM id is Dell or LSI
-            if (vdInfo.oemId) {
-                driveId.controllerVender = vdInfo.oemId.toLowerCase();
-            }
 
+            if (matchedBaseInfoKey ) {
+                // Base info key example:
+                //  "Drive /c0/e252/s0"
+                var match = matchedBaseInfoKey.match(/.*\/c(\d+)\/e(\d+)\/s(\d+)/);
+                matchedBaseInfo = driveBaseInfoList[matchedBaseInfoKey];
+                driveId.controllerId = controllerId; 
+                driveId.deviceIds = [matchedBaseInfo['DID']];
+                driveId.slotIds = matchedBaseInfoKey.split(' ').slice(-1);
+                driveId.size = matchedBaseInfo['Size'];
+                driveId.type = matchedBaseInfo['State'];
+                driveId.controllerVender = vendors[_.parseInt(controllerId)];
+                driveId.physicalDisks = [{
+                    protocol: matchedBaseInfo['Intf'],
+                    model: matchedBaseInfo['Model'],
+                    deviceId: matchedBaseInfo['DID'],
+                    slotId: match[3],
+                    size: matchedBaseInfo['Size'],
+                    type: matchedBaseInfo['Med'],
```

Comment:
I notice in many cases the code use a property without checking its existence. is there any case the a certain property might not appear in the catalog?